Gitweb:     http://git.kernel.org/linus/e38e2af1c57c3eb5211331a5b4fcaae0c4a2a918
Commit:     e38e2af1c57c3eb5211331a5b4fcaae0c4a2a918
Parent:     d9c2d5ac6af87b4491bff107113aaf16f6c2b2d9
Author:     Cliff Wickman <cpw@sgi.com>
AuthorDate: Thu Nov 19 17:12:43 2009 -0600
Committer:  Ingo Molnar <mingo@elte.hu>
CommitDate: Mon Nov 23 19:12:50 2009 +0100

    x86: SGI UV: Fix BAU initialization
    
    A memory mapped register that affects the SGI UV Broadcast
    Assist Unit's interrupt handling may sometimes be unintialized.
    
    Remove the condition on its initialization, as that condition
    can be randomly satisfied by a hardware reset.
    
    Signed-off-by: Cliff Wickman <cpw@sgi.com>
    Cc: <stable@kernel.org>
    LKML-Reference: <E1NBGB9-0005nU-Dp@eag09.americas.sgi.com>
    Signed-off-by: Ingo Molnar <mingo@elte.hu>
---
 arch/x86/kernel/tlb_uv.c |    4 +---
 1 files changed, 1 insertions(+), 3 deletions(-)

--- a/arch/x86/kernel/tlb_uv.c
+++ b/arch/x86/kernel/tlb_uv.c
@@ -817,10 +817,8 @@ static int __init uv_init_blade(int blad
 	 */
 	apicid = blade_to_first_apicid(blade);
 	pa = uv_read_global_mmr64(pnode, UVH_BAU_DATA_CONFIG);
-	if ((pa & 0xff) != UV_BAU_MESSAGE) {
-		uv_write_global_mmr64(pnode, UVH_BAU_DATA_CONFIG,
+	uv_write_global_mmr64(pnode, UVH_BAU_DATA_CONFIG,
 				      ((apicid << 32) | UV_BAU_MESSAGE));
-	}
 	return 0;
 }
 
